### DISCUSSION TITLE
Tutorial: How to deploy a MkDocs static site on Azure

### DISCUSSION THREAD

**Discussion by JC5** [squidfunk/mkdocs-material#2192] [Show and tell]
I noticed this today, and it took me some time to find the issue. Contrary to most (if not all) web servers, Azure static sites handles the following URL's as separate URL's:

1. https://docs.firefly-iii.org/csv
2. https://docs.firefly-iii.org/csv/

On most webservers nr.1 automatically redirects to nr.2 (see for example `mkdocs serve`) but Azure does not. The result is that most menu's give 404's when people visit the first URL, because the links are relative.

I managed [to solve](https://docs.firefly-iii.org/csv) this by adding a `<base href>` to the main template.

Azure static sites btw, is when you create an Azure storage bucket and enable it for static web hosting. Combined with a CDN you can use it to cheaply host static websites. 

I'm not sure if this is the best solution, but is it maybe an idea to mention it somewhere in the docs? Or maybe this ticket is enough :)

**Comment by squidfunk** [comment]
Thanks for sharing your solution! We try to keep the docs as concise as possible, so enumerating all possible quirks with different hosting providers is probably beyond the scope of this project. However, by providing your experience here, future users will have it easier to set up Material for MkDocs on Azure 🙏

**Comment by PBittinger** [comment]
Hi JCS.  I deployed a MKDOCS site to Azure (Azure Web App), and the search capability is not working.   
When I click into the search box, I get 'Initializing Search'.  The search works perfectly on the localhost version of the website.  I am using latest version of material and mkdocs.  Did you run into this issue when you deployed to Azure?   I have tried lots of different options in the yml file and nothing fixes it.

**Reply by PBittinger** [reply]
Sorry - I just found the answer on another post on github.  This is an issue in general for MKDOCS sites hosted on azure, not just material theme.  This fix worked for me - creating a web.config file.
https://github.com/mkdocs/mkdocs/issues/1468

**Reply by PBittinger** [reply]
Hi.  Did you create a yml file under github actions?  What is the issue you are having?

**Reply by Eduardadantas** [reply]
In my case, I am using Azure Repos.
I created the pipeline and it worked, but when I open the site it doesn't appear at all.
I believe that my problem is somewhere related to mkdocs. Since my pipeline worked.
I need to use flask to this??
